packages/protocol/scripts/truffle/upgrade.ts

```diff
@@ -161,14 +209,20 @@ module.exports = async (callback: (error?: any) => number) => {
         process.exit(0)
       }
 
-      contractsToUpgrade.forEach(
-        async ([Contract, Proxy]: [Truffle.Contract<any>, Truffle.Contract<ProxyInstance>]) => {
-          // tslint:disable-next-line:no-console
-          console.log('Updating', Contract.contractName)
-          const proxy = await Proxy.deployed()
-          await upgradeContract(Contract, proxy)
-        }
+      const lst = await Promise.all(
+        contractsToUpgrade.map(
+          async ([Contract, Proxy]: [Truffle.Contract<any>, Truffle.Contract<ProxyInstance>]) => {
+            // tslint:disable-next-line:no-console
+            console.log('Updating', Contract.contractName)
+            const proxy = await Proxy.deployed()
+            return upgradeContract(Contract, proxy)
+          }
+        )
       )
+      // tslint:disable-next-line:no-console
+      console.log('Generated proposal')
```

Comment:
What exactly is the structure of this proposal? Is it bytecode that can immediately be signed and submitted as a transaction? Does it have a human readable component?

Reply:
It should be a json file that can be submitted using the CLI tool.